### ISSUE TITLE
difference of ROIs

### ISSUE THREAD

**Issue by satra** [gablab/murfi2#11]
is there a way to implement a difference of roi value calculation? when one roi is background/wm it's variance could be low and hence z value can be really high. subtracting z values makes sense only if one is comparing task related rois.

in the other cases, one might want to z-ify the difference in activity between two mask regions.

**Comment by ohinds** [comment]
Maybe I don't understand what you are getting at. 

I think I disagree with the statement "subtracting z values makes sense only if one is comparing task related rois". It is true that a change that is lower in magnitude will produce a greater z-score change if the variance is lower, but that is an intended effect. The idea is that you figure out how variable a voxel is, then look at changes relative to this variance. If the z-score is really high, that means that a really high deviation occurred, relative to what was expected. If that should be different for non-task related regions, I'd need more explanation than what is above.

Also, are you suggesting taking a mean of intensity across an ROI, then subtracting that across ROIs? I don't think that's such a good idea, because the deviations from the mean intensity in different ROIs could be of wildly different magnitude (which is what the z-score approach is supposed to avoid)

**Comment by satra** [comment]
i think this is something we need to think about a bit. i'll give an example.

say roi1 = some task (some variance)
and roi2 = subset of wm (very little variance)

z_roi2 > z_roi1 primarily because of the reduce variance.

so if we are giving feedback as z_roi1 - z_roi2 (a negative number), a positive change in z_roi1 might mean moving towards 0. 

one of the reasons for thinking about taking the difference of raw values is that scanner related changes are likely to affect all regions, the difference will then eliminate any fluctuation or drift. 

i think this needs to be thought through a little more clearly.

**Comment by ohinds** [comment]
That may be so, but that is the uncommon case, and if you chose that route, most of the time raw_roi1 would be >> than raw_roi2, and would do basically nothing.

**Comment by satra** [comment]
i agree but z(raw_roi1 - raw_roi2) would probably take care of any non-task changes affecting both roi1/2.

**Comment by ohinds** [comment]
I think a better way to do this would be to track the difference in mean intensity over the entire volume relative to the first volume, then subtract out this difference from each voxel before computing the z-score.
